server/lib/nl/nl_variable.py

```diff
@@ -43,38 +57,82 @@ class VariableStore:
+  parts = sv_definition.split(",")
+  for part in parts:
+    k, v = part.split("=")
+    if k == "pt":
+      res.pt = v
+    elif k == "mp":
+      res.mp = v
+    elif k == "st":
+      res.st = v
     else:
-      result[svg] = svs
-  if svgs_l2:
-    child_svgs = []
-    svg2childsvgs = dc.property_values(svgs_l2, "specializationOf", False)
-    all_child_svgs = []
-    for svg, child_svgs in svg2childsvgs.items():
-      all_child_svgs.extend(child_svgs)
-    childsvg2svs = dc.property_values(all_child_svgs, "memberOf", False)
-    for svg, child_svgs in svg2childsvgs.items():
-      for child_svg in child_svgs:
-        result[svg].extend(childsvg2svs[child_svg])
-  return result
-
-
-def extend_svs(svs: List[str]):
-  """Extend all svs with siblings svs under the same svg.
+      res.pvs[k] = v
+  return res
+
+
+def parse_svg(svg_dcid: str) -> SVG:
+  res = SVG()
+  body = svg_dcid[len("dc/g/"):]
+  parts = body.split("_")
+  if len(parts) == 1:
+    return res
+  res.pt = parts[0]
+  for part in parts[1:]:
+    if "-" in part:
+      p, v = part.split("-")
+      p_lower = p.replace(p[0], p[0].lower(), 1)
+      res.pvs[p_lower] = v
+    else:
+      res.p = part.replace(part[0], part[0].lower(), 1)
+  return res
+
+
+def extend_svs(svs: Dict[str, List[str]]):
+  """Extend all svs with siblings svs.
   """
+  if not svs:
+    return {}
   sv2svgs = dc.property_values(svs, "memberOf", True)
   sv2svg = {sv: svg[0] for sv, svg in sv2svgs.items() if svg}
-  svg2children = dc.property_values(sv2svg.values(), "memberOf", False)
-  result = {}
+  svg2childsvs = {}
+  svginfo = dc.get_variable_group_info(list(sv2svg.values()), [])
+  for item in svginfo['data']:
+    svg2childsvs[item['node']] = item['info']['childStatVars']
+
+  res = {}
   for sv, svg in sv2svg.items():
-    result[sv] = svg2children.get(svg, [])
-  return result
+    res[sv] = []
+    svg_obj = parse_svg(svg)
+    sv_obj = None
+    for child_sv in svg2childsvs[svg]:
+      if child_sv['id'] == sv:
+        sv_obj = parse_sv(child_sv['definition'])
+        break
+    if not sv_obj:
+      continue
+    if len(svg_obj.pvs) == len(sv_obj.pvs):
+      # There are no direct siblings of this sv in the current svg.
+      # need to look for in-direct siblings
+      svg_parent = dc.property_values([svg], "specializationOf", True)[svg][0]
+      svg_siblings = dc.property_values([svg_parent], "specializationOf",
+                                        False)[svg_parent]
+      svg_siblings_info = dc.get_variable_group_info(svg_siblings, [])
+      for item in svg_siblings_info['data']:
+        for sv_info in item['info'].get('childStatVars', []):
+          curr_sv_obj = parse_sv(sv_info['definition'])
+          if curr_sv_obj.mp != sv_obj.mp:
+            continue
+          if curr_sv_obj.st != sv_obj.st:
+            continue
+          if curr_sv_obj.pt != sv_obj.pt:
+            continue
+          if len(curr_sv_obj.pvs) != len(sv_obj.pvs):
+            continue
```

Comment:
should we consider measurementDenominator too?  I think measurementQualifier being different maybe ok for peers.

Reply:
Done

Reply:
Do we need to handle md in comparison in extend_svs() ?

Reply:
Done